samza-core/src/main/java/org/apache/samza/storage/StorageRecovery.java

```diff
@@ -158,6 +155,7 @@ private void getContainerModels() {
    * Get the changelog streams and the storage factories from the config file
    * and put them into the maps
    */
+  @SuppressWarnings("unchecked")
```

Comment:
Instead of doing a blanket suppress for the whole method, could you please consider doing a suppression for the single line that needs to be suppressed? Then, if someone accidentally makes a change which should not be suppressed, then there will still be a warning for it.

Reply:
Updated to suppress the single line.